### PR TITLE
Fix push-manifests Makefile target

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

Third party `manifest-tool` is removed from calico/go-build:v0.90 in favor of the official `docker manifest` command. This change re-implements the push-manifests target.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by hjiawei** [comment]
Tested manually. The `push-manifests` target will be expanded to:

```bash
$ IMAGETAG=1.2.3 make push-manifests 
docker manifest create quay.io/tigera/operator:1.2.3 --amend quay.io/tigera/operator:1.2.3-amd64 --amend quay.io/tigera/operator:1.2.3-arm64 --amend quay.io/tigera/operator:1.2.3-ppc64le --amend quay.io/tigera/operator:1.2.3-s390x
```

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
